app/src/main/java/com/example/countryinfo/CountryPreviewFragment.kt

```diff
@@ -28,7 +28,9 @@ class CountryPreviewFragment : Fragment() {
 
     override fun onAttach(context: Context) {
         super.onAttach(context)
-        listener = context as MainActivity
+        if (context is MainActivity) {
+            listener = context
+        }
```

Comment:
Cover this with tests also. Please investigate possible solutions.

Reply:
Done.